src/View/SSViewer_Scope.php

```diff
@@ -308,6 +320,8 @@ public function next()
             // Item may be an array or a regular IteratorAggregate
             if (is_array($this->item)) {
                 $this->itemIterator = new ArrayIterator($this->item);
+            } elseif ($this->item instanceof Iterator) {
+                $this->itemIterator = $this->item;
             } else {
```

Comment:
I think it's worth adding this because:
1. there's already logic here to handle native arrays, even though previously those couldn't be used at all and now they're wrapped in `ArrayList` before we get here... but this allows for them to be handled correctly if they somehow sneak in
2. As with the array check, this allows iterators to be used correctly if they somehow sneak in. They shouldn't get here, but if they do it's better to just let them be iterated over rather than throw exceptions about it, IMO.